### PR TITLE
fix: consensus reads: correct handling of read orientation

### PR DESCRIPTION
**Pull request by FelixMoelder**
This PR introduces some changes to the handling of read orientations during consensus read calculation.
In its previous implementation records originating from the reverse strand where not written as reverse complement to the fastq-files. This has been fixed.

In addition, duplicate reads are grouped not only by identical CIGAR strings but also by read orientation.
This means grouping is performed across read pairs ensuring that all duplicate r1 records originate from the same strand as well as all r2 records.
Therefore, it is only necessary to store a strand information string for overlapping consensus reads where the mates are from different stands.
In all other cases, no strand information is saved. 

Edit:
An additional `EF`-tag will be added to overlapping consensus reads denoting that the read covers the entire fragment.

### CODE REVIEW

File: src/bam/collapse_reads_to_fragments/calc_consensus.rs [hunk 1]
```diff
@@ -93,26 +91,36 @@ impl<'a> CalcOverlappingConsensus<'a> {
...
             self.build_consensus_strand(&mut consensus_strand, consensus_seq[i], i);
         }
+        let mut tag_string = Vec::new();
+        match self.recs1()[0].read_pair_orientation().as_ref() {
+            "R1R2" | "R2R1" => {
+                consensus_seq = revcomp(consensus_seq);
+                consensus_qual.reverse();
+            }
+            "R1F2" | "F1R2" | "R2F1" | "F2R1" => {
+                tag_string = [tag_string, b"SI:Z:".to_vec(), consensus_strand, vec![b' ']].concat();
```
Comment (FelixMoelder) [inline src/bam/collapse_reads_to_fragments/calc_consensus.rs:101]: I don't like this kind of concatenation for `Vec<u8>` as I need to at whitespaces manually. Is there something equivalent for joining Vectors with a separator as for Strings?

Comment (TheLostLambda) [reply src/bam/collapse_reads_to_fragments/calc_consensus.rs:101]: Appears to exist for slices? https://doc.rust-lang.org/std/vec/struct.Vec.html#method.join

File: src/common.rs [hunk 2]
```diff
@@ -90,6 +93,8 @@ pub trait CalcConsensus<'a, R: SequenceRead> {
                 .unwrap())
                 .join(",")
         )
+        .as_bytes()
+        .to_vec()
```
Comment (FelixMoelder) [inline src/common.rs:97]: This format!-Macro does not really make sense as the resulting string is transformed to a byte vector afterwards.
I tried to directly work on byte-arrays but without success.
Any idea on how to refactor this as I do not want this to be merged.

**Comment by FelixMoelder** [comment]
Paired records will now be written to the correct fq-writer.
This means that if the first mate is tagged as `second in pair`, it will be written to the fq2-writer, while the second mate will be written to the fq1-writer.

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**